solutions/cses-1139.mdx

```diff
@@ -0,0 +1,117 @@
+---
+id: cses-1139
+source: CSES
+title: Distinct Colors
+author: Timothy Gao
+---
+
+## Solution 1 - Small to Large Merging
+
+See [here](/plat/merging#merging-data-structures).
+
+## Solution 2 - PURS
+
+Let us consider the Euler Tour of the tree. We flatten the tree into an array, where each node corresponds to a range in this array. Now, we are essentially tasked to find the number of distinct values in a range $N$ times (once for each node).
+
+Let us consider iterating through the Euler Tour array from left to right. 
+When we consider a node at index $i$ in the Euler Tour array, its subtree range will be $[j, i]$, where $j\leq i$.
+Now, let's focus on a single color. Notice that if there are multiple colors $c$ to the left of a certain index $i$,
+only $c$'s most rightmost occurrence is relevant. This is because any non-rightmost occurrence of $c$
+from $i$ must include the rightmost occurrence of $c$. More formally, 
+any segment $[l, r]$ must contains $i$ if $l <= i <= r$. 
```

Comment:
> More formally, 
any segment $[l, r]$ must contains $i$ if $l <= i <= r$. 

uh, is this necessary

Reply:
doesn't hurt

Reply:
grammar